woocommerce/woocommerce-ios
# [Woo POS] M2 Error: Checking order

Closes: #13479


## Description


Handle 'Checking order' stage errors.

<img width="448" alt="image" src="https://github.com/user-attachments/assets/441b3d51-d707-48ca-868f-8b9564499532">

⚠️  There're 2 differences from the designs:
1. No "Learn More" button. There's no further information to display when this error happens
2. No darker background. We should do this in a separate task since it affects other views as well (https://github.com/woocommerce/woocommerce-ios/issues/13549). We cannot just set a background color for message view, since `TotalsView` has its centering, spacing, and padding. We likely need to restructure `TotalsView` so cardReaderView would be placed inside a larger container, which would touch the leading, trailing, and top sides. When required, we can change the color of this container, otherwise cardReaderView placement would be unaffected.

## Solution

Error when validating order (`CollectOrderPaymentUseCaseError.couldNotRefreshOrde`r) needs to be presented separately, therefore:

1. created a new `validatingOrderErro`r message type
2 Mapped `CollectOrderPaymentUseCaseError.couldNotRefreshOrder` `paymentError` to `validatingOrderError` message type
3. Created a new message view sharing most of layout configuration with ReaderDisconnectedMessageView

## Testing information


1. Either use CharlesProxy/Proxyman to trigger an error, or make changes to `OrderState retrieveOrderRemotely` to pass a wrong `siteId` can trigger one of the errors.
2. Open POS
2. Connect Reader
3. Add Items to Cart
4. Check Out
5. Notice error appears after validating order

## Screenshots


https://github.com/user-attachments/assets/304b1afe-0dba-4915-b693-26e583dc6e2a

---
